pennylane/utils.py

```diff
@@ -323,6 +326,22 @@ def circuit2():
             + ",".join(string_reps)
         )
 
+    if qml.tape_mode_active():
+        for op in operation_list:
+            try:
+                qml.tape.QueuingContext.remove(op)
+            except KeyError:
+                pass
+
+        with qml.tape.QuantumTape() as tape:
+            for o in operation_list:
+                o.queue()
+                if o.inverse:
+                    o.inv()
```

Comment:
How come `o` may be queued twice here?

Reply:
Ah! This is a feature of the _new_ queuing system.

1. `o.queue()` will queue the object
2. `o.inv()` will _annotate_ the queue - it will look for the `o` entry, and leave an annotation saying 'this object should be inverted'.

So we need the object to be in the queue before we can invert it.